src/utils/confighandler.cpp

```diff
@@ -464,15 +464,17 @@ QVariant ConfigHandler::value(const QString& key) const
 
 QSet<QString>& ConfigHandler::recognizedGeneralOptions()
 {
+    static QList<QString> optionList = ::recognizedGeneralOptions.keys();
     static QSet<QString> options =
-      QSet<QString>::fromList(::recognizedGeneralOptions.keys());
+        QSet<QString>(optionList.begin(), optionList.end());
     return options;
 }
 
 QSet<QString>& ConfigHandler::recognizedShortcutNames()
 {
+    static QList<QString> nameList = ::recognizedShortcuts.keys();
     static QSet<QString> names =
-      QSet<QString>::fromList(recognizedShortcuts.keys());
+        QSet<QString>(nameList.begin(), nameList.end());
     return names;
```

Comment:
Leave this as it was. Older versions of Qt do not support the new constructors you added. Linux distros using older Qt versions won't compile.

@borgmanJeremy  tried the same thing in #2023. What actually needs to be done is add a comment that explains why the older constructors have to be used, and possibly a clang-tidy ignore rule. (not very familiar with clang-tidy so I'm not sure how this is done). Not sure if @borgmanJeremy was going to make this change himself.

Reply:
No, we need to change it the way I did with a macro. 

I'm trying to remove all deprecated Qt warnings so we can build against Qt6. The warning is coming from Qt directly, this method of initialization will be removed.

Reply:
I'll finish this change on my branch

Reply:
OK, I'll revert this part